{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import statsmodels.api as sm\n",
    "import statsmodels.formula.api as smf\n",
    "import seaborn as sns\n",
    "import statsmodels\n",
    "import os\n",
    "import sys\n",
    "from os import listdir\n",
    "from os.path import isfile, join\n",
    "import functools \n",
    "from matplotlib_venn import venn3\n",
    "from collections import defaultdict\n",
    "import matplotlib.pyplot as plt\n",
    "plt.rcParams['figure.figsize'] = [8, 6]\n",
    "plt.rcParams['figure.dpi'] = 600"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "expression_file = \"/data8t/mtx/scSNV/dataset_v2/GSE57872/data/quantification_stats/gene_expression_matrix/gene_expression_matrix_cpm.csv\"\n",
    "exp_profile = pd.read_csv(expression_file,index_col=0)\n",
    "reads_file = \"/data8t/mtx/scSNV/dataset_v2/GSE57872/data/quantification_stats/gene_expression_matrix/gene_expression_matrix.csv\"\n",
    "reads_profile = pd.read_csv(reads_file,index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "pvalue_cellline = pd.read_csv('cellline.csv',index_col=0)\n",
    "pvalue_patients = pd.read_csv('patients.csv',index_col=0)\n",
    "pvalue_all = pd.read_csv('all_gene_snv.csv',index_col=0)\n",
    "\n",
    "gene_patients = pd.read_csv('/data8t/mtx/scSNV/dataset_v2/GSE57872/data/processed_profile/GSE57872_final_cpm_MGH.csv', index_col=0)\n",
    "snv_patients = pd.read_csv('/data8t/mtx/scSNV/dataset_v2/GSE57872/data/processed_profile/GSE57872_final_snv_MGH.csv', index_col=0)\n",
    "\n",
    "gene_cellline = pd.read_csv('/data8t/mtx/scSNV/dataset_v2/GSE57872/data/processed_profile/GSE57872_final_cpm_CSC.csv', index_col=0)\n",
    "snv_cellline = pd.read_csv('/data8t/mtx/scSNV/dataset_v2/GSE57872/data/processed_profile/GSE57872_final_snv_CSC.csv', index_col=0)\n",
    "\n",
    "label_all = pd.read_csv('/data8t/mtx/scSNV/dataset_v2/GSE57872/data/processed_profile/GSE57872_cellline_patient_label.csv', index_col=0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>label</th>\n",
       "      <th>single_or_bulk</th>\n",
       "      <th>sample_title</th>\n",
       "      <th>subtype</th>\n",
       "      <th>description</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>run_accession</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>SRR1294493</th>\n",
       "      <td>MGH26</td>\n",
       "      <td>Single cell mRNA-seq</td>\n",
       "      <td>Single cell mRNA-seq_MGH26_A02</td>\n",
       "      <td>subtype: Pro</td>\n",
       "      <td>processed data file: GBM_data_matrix.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRR1294494</th>\n",
       "      <td>MGH26</td>\n",
       "      <td>Single cell mRNA-seq</td>\n",
       "      <td>Single cell mRNA-seq_MGH26_A03</td>\n",
       "      <td>subtype: Pro+Cla</td>\n",
       "      <td>processed data file: GBM_data_matrix.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRR1294496</th>\n",
       "      <td>MGH26</td>\n",
       "      <td>Single cell mRNA-seq</td>\n",
       "      <td>Single cell mRNA-seq_MGH26_A05</td>\n",
       "      <td>subtype: Pro+Cla</td>\n",
       "      <td>processed data file: GBM_data_matrix.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRR1294497</th>\n",
       "      <td>MGH26</td>\n",
       "      <td>Single cell mRNA-seq</td>\n",
       "      <td>Single cell mRNA-seq_MGH26_A06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Please note that this sample did not pass the ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRR1294498</th>\n",
       "      <td>MGH26</td>\n",
       "      <td>Single cell mRNA-seq</td>\n",
       "      <td>Single cell mRNA-seq_MGH26_A07</td>\n",
       "      <td>subtype: Pro</td>\n",
       "      <td>processed data file: GBM_data_matrix.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRR1295349</th>\n",
       "      <td>CSC8</td>\n",
       "      <td>Single cell mRNA-seq</td>\n",
       "      <td>Single cell mRNA-seq_CSC8_H06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>processed data file: GBM_data_matrix.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRR1295350</th>\n",
       "      <td>CSC8</td>\n",
       "      <td>Single cell mRNA-seq</td>\n",
       "      <td>Single cell mRNA-seq_CSC8_H07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>processed data file: GBM_data_matrix.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRR1295352</th>\n",
       "      <td>CSC8</td>\n",
       "      <td>Single cell mRNA-seq</td>\n",
       "      <td>Single cell mRNA-seq_CSC8_H09</td>\n",
       "      <td>NaN</td>\n",
       "      <td>processed data file: GBM_data_matrix.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRR1295354</th>\n",
       "      <td>CSC8</td>\n",
       "      <td>Single cell mRNA-seq</td>\n",
       "      <td>Single cell mRNA-seq_CSC8_H11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>processed data file: GBM_data_matrix.txt</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRR1295355</th>\n",
       "      <td>CSC8</td>\n",
       "      <td>Single cell mRNA-seq</td>\n",
       "      <td>Single cell mRNA-seq_CSC8_H12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>processed data file: GBM_data_matrix.txt</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>489 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               label        single_or_bulk                    sample_title  \\\n",
       "run_accession                                                                \n",
       "SRR1294493     MGH26  Single cell mRNA-seq  Single cell mRNA-seq_MGH26_A02   \n",
       "SRR1294494     MGH26  Single cell mRNA-seq  Single cell mRNA-seq_MGH26_A03   \n",
       "SRR1294496     MGH26  Single cell mRNA-seq  Single cell mRNA-seq_MGH26_A05   \n",
       "SRR1294497     MGH26  Single cell mRNA-seq  Single cell mRNA-seq_MGH26_A06   \n",
       "SRR1294498     MGH26  Single cell mRNA-seq  Single cell mRNA-seq_MGH26_A07   \n",
       "...              ...                   ...                             ...   \n",
       "SRR1295349      CSC8  Single cell mRNA-seq   Single cell mRNA-seq_CSC8_H06   \n",
       "SRR1295350      CSC8  Single cell mRNA-seq   Single cell mRNA-seq_CSC8_H07   \n",
       "SRR1295352      CSC8  Single cell mRNA-seq   Single cell mRNA-seq_CSC8_H09   \n",
       "SRR1295354      CSC8  Single cell mRNA-seq   Single cell mRNA-seq_CSC8_H11   \n",
       "SRR1295355      CSC8  Single cell mRNA-seq   Single cell mRNA-seq_CSC8_H12   \n",
       "\n",
       "                        subtype  \\\n",
       "run_accession                     \n",
       "SRR1294493         subtype: Pro   \n",
       "SRR1294494     subtype: Pro+Cla   \n",
       "SRR1294496     subtype: Pro+Cla   \n",
       "SRR1294497                  NaN   \n",
       "SRR1294498         subtype: Pro   \n",
       "...                         ...   \n",
       "SRR1295349                  NaN   \n",
       "SRR1295350                  NaN   \n",
       "SRR1295352                  NaN   \n",
       "SRR1295354                  NaN   \n",
       "SRR1295355                  NaN   \n",
       "\n",
       "                                                     description  \n",
       "run_accession                                                     \n",
       "SRR1294493              processed data file: GBM_data_matrix.txt  \n",
       "SRR1294494              processed data file: GBM_data_matrix.txt  \n",
       "SRR1294496              processed data file: GBM_data_matrix.txt  \n",
       "SRR1294497     Please note that this sample did not pass the ...  \n",
       "SRR1294498              processed data file: GBM_data_matrix.txt  \n",
       "...                                                          ...  \n",
       "SRR1295349              processed data file: GBM_data_matrix.txt  \n",
       "SRR1295350              processed data file: GBM_data_matrix.txt  \n",
       "SRR1295352              processed data file: GBM_data_matrix.txt  \n",
       "SRR1295354              processed data file: GBM_data_matrix.txt  \n",
       "SRR1295355              processed data file: GBM_data_matrix.txt  \n",
       "\n",
       "[489 rows x 5 columns]"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "label_all"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Filter out all-non-zero genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
